### DISCUSSION TITLE
How to optimize startup time?

### DISCUSSION THREAD

**Discussion by hucancode** [medusajs/medusa#2600] [Q&A] [answered]
Hello,
I deployed Medusa server on a GCP Cloud Run instance. The machine get shut down when there are no traffic and get started when there are request comming. It works fine except the start up time is a little bit longer than I expected (13s~14s).

![image](https://user-images.githubusercontent.com/15852849/201550927-6d9d08a4-d4e6-4a84-88fd-a6ede0a7f581.png)

I have:
- A clean new Medusa project, sample seed data with `npm run seed`
- Postgres DB and Redis DB is in a GCP Compute Engine instance
- Nextjs storefront, ISR, result get cached for 20mins before revalidation. I see my Cloud Run instances doing cold boot quite alot.
- I set `MEDUSA_DISABLE_TELEMETRY=1` but still seeing log message saying I can disable telemetry with `medusa telemetry --disable` (not sure why).
- Here is my Dockerfile if that relevant
```Dockerfile
FROM node:18-alpine
WORKDIR /app/medusa
COPY package.json .
RUN npm install -g @medusajs/medusa-cli@latest
RUN npm install --omit=dev
COPY . .
CMD medusa start
```

What can I do to reduce the start up time?

**Comment by srindom** [comment, marked as answer]
Medusa is built with the assumption that the server is always on - we are exploring some options for making Medusa more suitable in serverless environments, which includes reducing the cold start time, but there is no timeline on when work like that would be ready. Sorry I can't be of better help right now :(

**Comment by adrien2p** [comment]
With the next release, the cold start time should be improved. Let us know when you gett the release how much time it takes. When i lunch the server it takes around 3-4 sec

**Reply by hucancode** [reply]
Sure, definitely will try. But what version are you refering to? 1.6.4 or 1.7.x?

**Reply by adrien2p** [reply]
👋 can you try with version 1.6.5+ and let me know

**Reply by adrien2p** [reply]
Just a question, is your start time including the npm install etc?

**Reply by hucancode** [reply]
@adrien2p no, just `medusa start`

**Comment by hucancode** [comment]
@adrien2p Thank you for the fix Adrien, on my side generally there are ~ 1s faster startup. But it's still over 13s on the cloud. Not sure why but it's only 4~5s for dev mode on my local machine I expect it to be at most double that in real world. I think there are some fundamental problem with my setup.
